design/ceph-volume-provisioning.md

```diff
@@ -0,0 +1,137 @@
+# ceph-volume OSD Provisioning
+
+**Targeted for v0.9**
+
+Provisioning OSDs today is done directly by Rook. This needs to be simplified and improved by building
+on the functionality provided by the `ceph-volume` tool that is included in the ceph image.
+
+## Legacy Design
+As Rook is implemented today, the provisioning has a lot of complexity around:
+- Partitioning of devices for bluestore
+- Partitioning and configuration of a `metadata` device where the WAL and DB are placed on a different device from the data
+- Support for both directories and devices
+- Support for bluestore and filestore
+
+Since this is mostly handled by `ceph-volume` now, Rook should replace its own provisioning code and rely on `ceph-volume`.
+
+## ceph-volume Design
+
+`ceph-volume` is a CLI tool included in the `ceph/ceph` image that will be used to configure and run Ceph OSDs.
+`ceph-volume` will replace the OSD provisioning mentioned previously in the legacy design.
+
+At a high level this flow remains unchanged from the flow in the [one-osd-per-pod design](dedicated-osd-pod.md#create-new-osds).
+No new jobs or pods need to be launched from what we have today. The sequence of events in the OSD provisioning will be the following.
+- The cluster CRD specifies what nodes/devices to configure with OSDs
+- The operator starts a provisioning job on each node where OSDs are to be configured
+- The provisioning job:
+  - Detects what devices should be configured
+  - Calls `ceph-volume lvm batch` to prepare the OSDs on the node. A single call is made with all of the devices unless more specific settings are included for LVM and partitions.
```

Comment:
So does `ceph-volume` create the "LVMs" or does the user need to create them upfront?
Also can existing "LVMs" be used? E.g., a user has LVM for OS and a "partition" for data.

Just to document it, I had those questions till I read the "New Features" section. These questions have been answered by the section.